NAA_Analysis.py

```diff
@@ -300,30 +288,43 @@ def NAA_table(candidates, energy, branching_ratios, isotopes, half_lives,
                                         sorted_info[i][0],
                                         sorted_info[i][2],
                                         sorted_info[i][3]])
-        # sorted by energy
-        data_e[candidates[i]] = np.array([energy[i], isotopes[i],
-                                          half_lives[i], branching_ratios[i],
-                                          fractions[i], uncertainty[i]])
 
     table_headers = []
     table_headers.append('Energy [keV]')
     table_headers.append('Isotope')
     table_headers.append('Mass Fraction')
     table_headers.append('Fraction Unc.')
 
-    table_headers_e = []
-    table_headers_e.extend(['Energy [keV]', 'Isotope', 'Half-life [s]',
-                            'Branching Ratio', 'Mass Fraction',
-                            'Fraction Unc.'])
-
     frame = pd.DataFrame(data, index=table_headers)
     frame = frame.T
+    frame.to_csv('NAA_Results_Isotope.csv')
+
+
+def create_cache(candidates, energy, branching_ratios, isotopes, half_lives,
+                 mass_number, isotopic_abundance, cross_section):
+    """
+    create_cache creates an offline database containing all found isotopes
+    along with relevant information needed to calculate concentrations using
+    the NAA weight fraction formula.
+    """
+    data_e = {}
+    for i in range(len(candidates)):
+        # sorted by energy
+        data_e[candidates[i]] = np.array([energy[i], isotopes[i],
+                                          half_lives[i], branching_ratios[i],
+                                          mass_number[i],
+                                          isotopic_abundance[i],
+                                          cross_section[i]])
+    table_headers_e = []
+    table_headers_e.extend(['Energy [keV]', 'Isotope', 'Half-life [s]',
+                            'Branching Ratio', 'Mass Number',
+                            'Isotopic Abundance', 'Cross Section [b]'])
```

Comment:
In Python there is no point in initializing a list as `[]` before adding contents. In fact, it is slower because when you add contents it will need to re-allocate memory for it.

Just do something like
```
table_headers_e = ['Energy [keV]', 'Isotope', 'Half-life [s]',
+                            'Branching Ratio', 'Mass Number',
+                            'Isotopic Abundance', 'Cross Section [b]']
```